design-proposals/passt-feature-gate-iface-api-deprecation.md

```diff
@@ -0,0 +1,73 @@
+## Overview
+Following the emerging [Kubevirt Network Binding Plugin API](https://docs.google.com/document/d/13fLaqZuM8El_CT5XzxPv-1epKsB1-Spv_jtXveEkkdQ/edit#heading=h.z27rudfpwy44)
+and introduction of Passt network binding plugin [2](https://github.com/kubevirt/kubevirt/pull/10425) [3](https://github.com/kubevirt/kubevirt/pull/10621), 
+it's now possible to remove Passt network binding implementation from Kubevirt core.
+
+This doc outlines deprecation plan for Passt network binding, interface API and  feature gate.
+
+# Motivation
+Removing Passt network binding implementation from Kubevirt core and use Passt network binding plugin instead, 
+enable reduce the maintenance burden.
+Having Passt binding code outside Kubevirt's core may ease contribution process as there is no need to be familiar 
+with [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt) core code.
+
+## Goals
+- Deprecate Passt network interface API.
+- Remove Passt feature gate.
+- Remove Passt network binding from Kubevirt core.
+  
+## Non Goals
+
+## User Stories
+
+## Repos
+[kubevirt/kubevirt](https://github.com/kubevirt/kubevirt):
+- [Passt binding plugin hook sidecar](https://github.com/kubevirt/kubevirt/tree/main/cmd/sidecars/network-passt-binding)
+- [Passt binding plugin CNI](https://github.com/kubevirt/kubevirt/tree/main/cmd/cniplugins/passt-binding)
+
+# Design
+Deprecating Passt feature-gate and interface API and replacing it with Passt binding plugin
+require some changes users should be aware of and prepared for.
+
+Deprecating Passt binding interface API and feature-gate shall be done in the following 
+phases.
+
+## Phase 1
+Announce passt feature-gate and interface API deprecation in kubevirt-dev mailing list.
+This will give users some time to prepare for changes in Kubevirt next release v1.2.0. 
+
+Kubevirt next release v1.2.0 should have the following changes:
+- Passt feature gate shall set as deprecated
+  When enabled, a warning shall be raised saying the feature gate is deprecated, will be removed in next release, 
+  and network binding plugin API shall be used instead.
+
+- VM whose spec specifies passt interface (`spec.domain.devices.interface.[].passt`) will be created with 
+  Passt network binding hook sidecar and CNI, in a way Kubevirt core implementation is not executed.
+
+  This will require users to set the following Passt binding plugin prequisites in advance:
```

Comment:
If the user has to add the sidecar and the NAD to the kubevirt CR anyway, I don't see the point of this intermediate state. I would skip phase1.

Reply:
We can make this requirement less harsh by having additional phase that allow using both Passt interface API and network binding plugin API:
When Passt interface API is used:
- Kubevirt will set default image for passt hook sidecar image if non is registered (similar to slirp)
- Kubevirt core implementation will be executed only to configure the pod network namespace (what passt CNI does).
Alternatively, the user could still use the network binding API by registering Passt hook sidecar image and NAD and install Passt binding CNI.

This comes with maintenance cost in the period Passt interface API is supported.